internal/csi-addons/rbd/replication.go

```diff
@@ -903,7 +903,14 @@ func (rs *ReplicationServer) GetVolumeReplicationInfo(ctx context.Context,
 	lastSyncInfo, err := remoteStatus.GetLastSyncInfo(ctx)
 	if err != nil {
 		if errors.Is(err, rbderrors.ErrLastSyncTimeNotFound) {
-			return nil, status.Errorf(codes.NotFound, "failed to get last sync info: %v", err)
+			// log error and return response so that we can surface replication status
+			// and message in csi-addons
+			log.ErrorLog(ctx, "failed to get last sync info: %v", err)
+
+			return &replication.GetVolumeReplicationInfoResponse{
+				Status:        replication.GetVolumeReplicationInfoResponse_UNKNOWN,
+				StatusMessage: "sync info not found",
+			}, nil
```

Comment:
Initially there was a `codes.NotFound`. The way I read [the error codes from the spec](https://github.com/csi-addons/spec/tree/main/replication#error-scheme-5), `FAILED_PRECONDITION` would be more appropriate. No need to fake an error in a `GetVolumeReplicationInfoResponse`.